modules/navigation/godot_navigation_server.cpp

```diff
@@ -228,7 +228,11 @@ Vector<Vector3> GodotNavigationServer::map_get_path(RID p_map, Vector3 p_origin,
 	const NavMap *map = map_owner.get_or_null(p_map);
 	ERR_FAIL_COND_V(map == nullptr, Vector<Vector3>());
 
-	return map->get_path(p_origin, p_destination, p_optimize, p_navigation_layers);
+	Vector<int32_t> path_types = Vector<int32_t>();
+	TypedArray<RID> path_rids = TypedArray<RID>();
+	Vector<int64_t> path_owners = Vector<int64_t>();
+
+	return map->get_path(p_origin, p_destination, p_optimize, p_navigation_layers, path_types, path_rids, path_owners);
```

Comment:
Right now I'm using references here, which avoids the chance of passing `nullptr` and having to sprinkle `if` statements throughout `NavMap::get_path`.  However, it does force me to collect the metadata here for the "legacy" `map_get_path` implementation to avoid breaking compatibility, only to throw that metadata away.

Those with more experience in C++, would it be better to pass `nullptr` here and add checks throughout `NavMap::get_path`?  Are there any clever compiler optimizations that might influence the approach I should take here?

Reply:
Don't upgrade / support the old `map_get_path()` function, leave it alone, it is not worth it.

Reply:
No user facing changes have been made, but I did go ahead and convert the references to pointers in order to avoid any allocation overhead due to metadata collection.

This method should now perform basically the same as it did before (only theoretical cost would be some branch instructions, which the compiler will probably optimize out).